vpr/src/noc/noc_traffic_flows.h

```diff
@@ -0,0 +1,307 @@
+     * @param associated_router_id A ClusterblockId that represents a
+     *                             router block. 
+     * @param router_associated_traffic_flows A datastructure that stores
+     * a vector of traffic flows for a given router block where the traffic
+     * flows have the router as a source or sink within the flow.
+     *                                        
+     */
+    void add_traffic_flow_to_associated_routers(NocTrafficFlowId traffic_flow_id, ClusterBlockId associated_router_id, std::unordered_map<ClusterBlockId, std::vector<NocTrafficFlowId>>& router_associated_traffic_flows);
+
+    /**
+     * @brief Given a router block in the clustered netlist, store it
+     * internally.
+     * 
+     * @param router_block_id A unique identifier that rerpesents a router block in the clustered netlist. This id will be stored.
+     */
+    void add_router_to_block_set(ClusterBlockId router_block_id);
+
+  public:
+    NocTrafficFlows();
+
+    //getters
+
+    /**
+     * @brief Given a unique id of a traffic flow (t_noc_traffic_flow)
+     * retrieve it from the vector of all traffic flows in the design. The
+     * retrieved traffic flow cannot be modified but can be used to
+     * retireve information such as the routers involved.
+     * 
+     * @param traffic_flow_id The unique identifier (NocTrafficFlowId)
+     * of the traffic flow to retrieve.
+     * @return const t_noc_traffic_flow& The traffic flow represented by
+     * the provided identifier.
+     */
+    const t_noc_traffic_flow& get_single_noc_traffic_flow(NocTrafficFlowId traffic_flow_id) const;
+
+    /**
+     * @brief Get a vector of all traffic flows that have a given router
+     * block in the clustered netlist as the source (starting point) in the 
+     * flow.
+     * 
+     * @param source_router_id A unique identifier that represents the
+     * a router block in the clustered netlist. This router block will
+     * be the source router in the retrieved traffic flow.
+     * @return const std::vector<NocTrafficFlowId>* A vector of traffic 
+     * flows that have the input router block parameter as the source in the
+     * flow.
+     */
+    const std::vector<NocTrafficFlowId>* get_traffic_flows_associated_to_source_router(ClusterBlockId source_router_id) const;
+
+    /**
+     * @brief Get a vector of all traffic flows that have a given router
+     * block in the clustered netlist as the sink (end point) in the 
+     * flow.
+     * 
+     * @param sink_router_id A unique identifier that represents the
+     * a router block in the clustered netlist. This router block will
+     * be the sink router in the retrieved traffic flow.
+     * @return const std::vector<NocTrafficFlowId>* A vector of traffic 
+     * flows that have the input router block parameter as the sink in the
+     * flow.
+     */
+    const std::vector<NocTrafficFlowId>* get_traffic_flows_associated_to_sink_router(ClusterBlockId sink_router_id) const;
```

Comment:
If you wind up merging the two source/sink unordered_maps, then probably these functions merge too.

Reply:
Merged into a single function